vector/src/main/java/im/vector/adapters/MediasPreviewAdapter.java

```diff
@@ -0,0 +1,149 @@
+import android.view.MotionEvent;
+import android.view.View;
+import android.view.ViewGroup;
+import android.webkit.WebView;
+import android.widget.ImageView;
+import android.widget.TextView;
+import android.widget.VideoView;
+
+import org.matrix.androidsdk.data.RoomMediaMessage;
+
+import java.util.List;
+
+import im.vector.R;
+
+public class MediasPreviewAdapter extends RecyclerView.Adapter<MediasPreviewAdapter.ImageItemViewHolder> {
+
+    List<RoomMediaMessage> mImagePreviewList;
+
+    public MediasPreviewAdapter(List<RoomMediaMessage> imagePreviewList) {
+        mImagePreviewList = imagePreviewList;
+    }
+
+    public static class ImageItemViewHolder extends RecyclerView.ViewHolder {
+        ImageView imagePreview;
+
+        ImageItemViewHolder(View itemView) {
+            super(itemView);
+            imagePreview = (ImageView) itemView.findViewById(R.id.image_preview);
+        }
+    }
+
+    @Override
+    public int getItemCount() {
+        // Return the amount of items in the list.
+        return mImagePreviewList.size();
+    }
+
+    @Override
+    public ImageItemViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        // Create a new CardItem when needed.
+        View view = LayoutInflater.from(parent.getContext())
+                .inflate(R.layout.image_preview, parent, false);
+        ImageItemViewHolder imageItemViewHolder = new ImageItemViewHolder(view);
+        return imageItemViewHolder;
+    }
+
+    @Override
+    public void onBindViewHolder(ImageItemViewHolder holder, int position) {
+        final Context context = holder.imagePreview.getContext();
+        String mimeType = mImagePreviewList.get(position).getMimeType(context);
+        Uri uri = mImagePreviewList.get(position).getUri();
+
+        if (mimeType != null) {
+            if (mimeType.startsWith("image")) {
+                holder.imagePreview.setImageURI(uri);
+            } else if (mimeType.startsWith("video")) {
+                MediaMetadataRetriever mediaMetadataRetriever = new MediaMetadataRetriever();
+                mediaMetadataRetriever.setDataSource(context, uri);
+                Bitmap bitmap = mediaMetadataRetriever.getFrameAtTime();
+                holder.imagePreview.setImageBitmap(bitmap);
+            } else {
+                holder.imagePreview.setImageDrawable(ContextCompat.getDrawable(context, R.drawable.filetype_attachment));
+                holder.imagePreview.setScaleType(ImageView.ScaleType.FIT_CENTER);
+            }
+        }
+
+        final RoomMediaMessage mediaMessage = mImagePreviewList.get(position);
+
+        // Change to the image that is clicked on.
+        holder.itemView.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                ViewGroup viewGroup = (ViewGroup) view.getParent().getParent();
+                ImageView imagePreview = (ImageView) viewGroup.findViewById(R.id.image_previewer);
```

Comment:
It's not nice to bind this View here. They have to be bound in the Activity they belong to.
Instead notify the Activity of the click event with a listener to provide the clicked item position